plugins/modules/azure_rm_storageaccount.py

```diff
@@ -1303,15 +1373,54 @@ def failover_account(self):
     def update_network_rule_set(self):
         if not self.check_mode:
             try:
-                parameters = self.storage_models.StorageAccountUpdateParameters(network_rule_set=self.network_acls)
-                self.storage_client.storage_accounts.update(self.resource_group,
-                                                            self.name,
-                                                            parameters)
+                models = self.storage_models
+                nacls = self.network_acls or {}
+
+                vnet_rules = None
+                if nacls.get('virtual_network_rules'):
+                    vnet_rules = [
+                        models.VirtualNetworkRule(
+                            virtual_network_resource_id=r.get('id'),
+                            action=r.get('action', 'Allow')
+                        ) for r in (nacls.get('virtual_network_rules') or [])
+                    ]
+
+                ip_rules = None
+                if nacls.get('ip_rules'):
+                    ip_rules = [
+                        models.IPRule(
+                            ip_address_or_range=r.get('value'),
+                            action=r.get('action', 'Allow')
+                        ) for r in (nacls.get('ip_rules') or [])
+                    ]
+
+                resource_access_rules = None
+                # IMPORTANT: when desired list is [], send [] (not None) to clear rules
+                if nacls.get('resource_access_rules') is not None:
+                    resource_access_rules = [
+                        models.ResourceAccessRule(
+                            resource_id=r.get('resource_id'),
+                            tenant_id=r.get('tenant_id')
+                        ) for r in (nacls.get('resource_access_rules') or [])
+                    ]
+
+                nrs = models.NetworkRuleSet(
+                    bypass=nacls.get('bypass'),
+                    default_action=nacls.get('default_action', 'Allow'),
+                    virtual_network_rules=vnet_rules,
+                    ip_rules=ip_rules,
+                    resource_access_rules=resource_access_rules
+                )
+
+                parameters = self.storage_models.StorageAccountUpdateParameters(network_rule_set=nrs)
+                self.storage_client.storage_accounts.update(self.resource_group, self.name, parameters)
             except Exception as exc:
-                self.fail("Failed to update account type: {0}".format(str(exc)))
+                self.fail("Failed to update network ACLs: {0}".format(str(exc)))
 
-    def sort_list_of_dicts(self, rule_set, dict_key):
```

Comment:
This is not necessary as we can just rely on the "resource_id" for sorting, which is unique across tenants.